doc/developer/design/20230615_webhook_source.md

```diff
@@ -161,17 +164,101 @@ request headers. An issue though is everyone does this just a little bit differe
 * Buildkite signature = HMAC of "#{timestamp}.#{body}"
 * GitHub signature = "body=" + HMAC of {body}
 
-As such we'll need to support some custom logic for validation. What we can do is support a
-`VALIDATE USING` statement that accepts only a single scalar expression. This expression will
-be provided the request `headers` as `map[text] -> text`, and the `body` as `text` _regardless of
-whether or not the source has `INCLUDE HEADERS` or what `FORMAT BODY` is specified._ For example:
+As such we'll need to support some custom logic for validation. What we can do
+is support a `CHECK` constraint, like [PostgreSQL supports on
+tables][pg-check-constraint], which validates the incoming HTTP request
+by executing a Boolean-valued scalar expression. As an example:
 
 ```
-VALIDATE USING (
-  headers['X-Signature'] = hmac('sha256', SECRET webhook_secret, body)
+CHECK (
+  WITH (
+    SECRETS (db.schema.webhook_secret),
+    HEADERS,
+    BODY
+  )
+  headers['x-signature'] = hmac('sha256', webhook_secret, body)
 )
 ```
 
+The full syntax for the `CHECK` constraint is as follows:
+
+```ebnf
+check-option = 'CHECK' '('
+    ['WITH' '(' check-with (',' check-with)* ')']
+    scalar-expr
+')' ;
+
+check-with = (
+    'SECRETS' '(' check-with-secret (',' check-with-secret)* ')' |
+    'HEADERS' ['AS' ident] [BYTES] |
+    'BODY' ['AS' ident] [BYTES]
```

Comment:
This changes the behavior of `CHECK` from providing `headers` and `body` by default to requiring these keywords be specified. I've gone back and forth on this a couple of times, because by default I would assume users want headers and body included, so by requiring these keywords it adds verbosity, _but_ when looking at the syntax it does make the statement explicit which is kind of good. For example, 
```
CREATE SOUCE webhook IN CLUSTER foo FROM WEBHOOK
  BODY FORMAT TEXT
  CHECK (
    WITH (HEADERS, BODY)
    SELECT headers->>'signature' = sha256hash(body)
  )
```
doesn't feel too bad?

Reply:
Yeah, I went back and forth too as I was drafting up this PR, and ultimately came down on the side of being explicit. It's very easy later if we want to expose `headers` and `body` by default—that's a backwards compatible change—but the reverse is not!

Reply:
> It's very easy later if we want to expose headers and body by default—that's a backwards compatible change—but the reverse is not!

Very true! Let's go with it then